snuba/web/query.py

```diff
@@ -108,9 +108,12 @@ def _run_query_pipeline(
     ):
         metrics.increment("sample_without_turbo", tags={"referrer": request.referrer})
 
-    query_runner = partial(
-        _run_and_apply_column_names, timer, query_metadata, request.referrer,
-    )
+    if request.settings.get_dry_run():
+        query_runner = _dry_run_query_runner
```

Comment:
I would also skip the query recording phase in `parse_and_run_query` if you are in dry run mode, otherwise we would impact both querylog and our metrics